fedmsg/consumers/__init__.py

```diff
@@ -226,11 +227,30 @@ def _make_query(page=1):
                         break
 
     def validate(self, message):
-        """ This needs to raise an exception, caught by moksha. """
+        """
+        Validate the message before the consumer processes it.
+
+        This needs to raise an exception, caught by moksha.
+
+        Args:
+            message (dict): The message as a dictionary. This must, at a minimum,
+                contain the 'topic' key with a unicode string value and 'body' key
+                with a dictionary value.
+
+        Raises:
+            RuntimeWarning: If the message is not valid.
```

Comment:
Would ```ValueError``` make sense instead of RuntimeWarning? If not, what about ```RuntimeError``` instead of ```RuntimeWarning```?

Also, you could also document the potential encoding exception if the message fails to decode with the UTF-8 codec.